test/jdk/java/nio/file/Files/Links.java

```diff
@@ -48,6 +48,19 @@ static void assertTrue(boolean okay) {
      * Exercise createSymbolicLink and readLink methods
      */
     static void testSymLinks(Path dir) throws IOException {
+        // Check for FileAlreadyExistsException. This must be done
+        // before the check of whether symlinks are supported.
+        Path pathLink = Files.createTempFile("link", null);
+        Path pathTarget = Files.createTempFile("target", null);
+        try {
+            Files.createSymbolicLink(pathLink, pathTarget);
+            throw new RuntimeException("Symbolic link created");
+        } catch (FileAlreadyExistsException expected) {
+        } finally {
+            Files.deleteIfExists(pathLink);
+            Files.deleteIfExists(pathTarget);
+        }
+
```

Comment:
I think it's confusing to have a test for sym links before testing if sym links are supported. I think it's okay to not include tests that target very specific cases like Windows "developer mode" but a test that the optional specific exception is thrown is okay, it just feels like it should be just one of the tests, rather than done before knowing the test can run, e.g. try this on a non-NTFS system and I assume createSymbolicLink will fail and cause the test to fail.

Reply:
Logically I don't like the ordering either but without this happening first the error was not reproduced on the pre-PR code.

Reply:
Perhaps it would be better to revert the test change altogether and label the issue `noreg-hard`?